File: demos/arrangement_demo.py
# arrangement_demo.py

# NOTE: This is a conceptual demo based on current bindings

from cgalpy import *

def demo_arrangement():
    print("Creating arrangement...")

    arr = Arrangement_2()

    print("Arrangement created")

    # Example usage (depending on bindings availability)
    print("Iterating vertices:")
    for v in arr.vertices():
        print("Vertex degree:", v.degree())
        print("Is isolated:", v.is_isolated())

    print("\nIterating faces:")
    for f in arr.faces():
        print("Number of holes:", f.number_of_holes())

    print("\nIterating halfedges:")
    for e in arr.halfedges():
        print("Source:", e.source())
        print("Target:", e.target())

if __name__ == "__main__":
    demo_arrangement()

File: demos/arrangement_demo.cpp
// NOTE: Demo is being actively set up.
// Dependencies (Boost, GMP, MPFR) are currently being configured.
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Arr_segment_traits_2.h>
#include <CGAL/Arrangement_2.h>

#include <iostream>

typedef CGAL::Exact_predicates_exact_constructions_kernel Kernel;
typedef CGAL::Arr_segment_traits_2<Kernel> Traits_2;
typedef CGAL::Arrangement_2<Traits_2> Arrangement_2;
typedef Traits_2::Point_2 Point_2;
typedef Traits_2::X_monotone_curve_2 Segment_2;

int main() {
    Arrangement_2 arr;

    // Insert segments
    CGAL::insert(arr, Segment_2(Point_2(0, 0), Point_2(5, 5)));
    CGAL::insert(arr, Segment_2(Point_2(0, 5), Point_2(5, 0)));

    std::cout << "Arrangement created!" << std::endl;

    // Print counts
    std::cout << "Number of vertices: " << arr.number_of_vertices() << std::endl;
    std::cout << "Number of edges: " << arr.number_of_edges() << std::endl;
    std::cout << "Number of faces: " << arr.number_of_faces() << std::endl;

    // Iterate vertices
    std::cout << "\nVertices:\n";
    for (auto vit = arr.vertices_begin(); vit != arr.vertices_end(); ++vit) {
        std::cout << "Degree: " << vit->degree() << std::endl;
    }

    return 0;
}